pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -39,6 +40,91 @@ func newReconciler(scope *machineScope) *Reconciler {
 	}
 }
 
+var (
+	supportedGpuTypes = map[string]string{
+		"nvidia-tesla-k80":  "NVIDIA_K80_GPUS",
+		"nvidia-tesla-p100": "NVIDIA_P100_GPUS",
+		"nvidia-tesla-v100": "NVIDIA_V100_GPUS",
+		"nvidia-tesla-a100": "NVIDIA_A100_GPUS",
+		"nvidia-tesla-p4":   "NVIDIA_P4_GPUS",
+		"nvidia-tesla-t4":   "NVIDIA_T4_GPUS",
+	}
+)
+
+func containsString(sli []string, str string) bool {
+	for _, elem := range sli {
+		if elem == str {
+			return true
+		}
+	}
+	return false
+}
+
+// machineTypeAcceleratorCount represents nvidia-tesla-A100 GPUs which are only compatible with A2 machine family
+func (r *Reconciler) checkQuota(machineTypeAcceleratorCount int64) error {
+	region, err := r.computeService.RegionGet(r.projectID, r.providerSpec.Region)
+	if err != nil {
+		return fmt.Errorf("failed to get region via compute service: %v", err)
+	}
+	quotas := region.Quotas
+	var guestAccelerators = []*v1beta1.GCPAcceleratorConfig{}
+	// When the machine type has associated accelerator instances (A2 machine family), accelerators will be nvidia-tesla-A100s.
+	// Additional guest accelerators are not allowed so ignore the providerSpec GuestAccelerators.
+	if machineTypeAcceleratorCount != 0 {
+		guestAccelerators = append(guestAccelerators, &v1beta1.GCPAcceleratorConfig{AcceleratorType: "nvidia-tesla-a100", AcceleratorCount: machineTypeAcceleratorCount})
+	} else {
+		guestAccelerators = r.providerSpec.GuestAccelerators
+	}
+	// validate zone and then quota
+	accelerator := guestAccelerators[0] // guestAccelerators slice cannot be longer than 1
```

Comment:
Why is this limited to 1? Is this a GCP limitation, if so, where is this written in the docs?

Reply:
I did not find it the docs, however gcp web console does not even have an option for attaching more then 1 type of GPU. And I tried to create an instance with 2 different guest accelerators through the api, and got an googleapi error saying  guestaccelerators is of size 2 which is too long.